src/client.rs

```diff
@@ -300,90 +210,141 @@ pub mod future {
                 }
-            };
+            }
             fn panic(canceled: futures::Canceled) -> io::Error {
                 unreachable!(canceled)
             }
-            ConnectFuture { inner: future::Either::B(rx.map_err(panic as fn(_) -> _).flatten()) }
+            rx.map_err(panic as _).flatten()
         }
     }
+
+    type ResponseFuture<Req, Resp, E> =
+        futures::AndThen<futures::MapErr<
+        futures::Map<<BindClient<Req, Resp, E> as Service>::Future,
+                     fn(WireResponse<Resp, E>) -> Result<Resp, ::Error<E>>>,
+            fn(io::Error) -> ::Error<E>>,
+                     Result<Resp, ::Error<E>>,
+                     fn(Result<Resp, ::Error<E>>) -> Result<Resp, ::Error<E>>>;
+    type BindClient<Req, Resp, E> =
+        <Proto<Req, Result<Resp, WireError<E>>>
+            as ProtoBindClient<Multiplex, StreamType>>::BindClient;
 }
 
 /// Exposes a trait for connecting synchronously to servers.
 pub mod sync {
-    use client::future::Connect as FutureConnect;
-    use futures::{Future, future};
+    use super::Reactor;
     use serde::{Deserialize, Serialize};
+    use std::fmt;
     use std::io;
     use std::net::ToSocketAddrs;
-    use super::{Client, Options};
+    use super::Options;
+    use super::future::{Client as FutureClient, ClientExt as FutureClientExt};
+    use tokio_core::reactor;
+    use tokio_service::Service;
     use util::FirstSocketAddr;
 
-    /// Types that can connect to a server synchronously.
-    pub trait Connect: Sized {
+    #[doc(hidden)]
+    pub struct Client<Req, Resp, E>
+        where Req: Serialize + 'static,
+              Resp: Deserialize + 'static,
+              E: Deserialize + 'static
+    {
+        inner: FutureClient<Req, Resp, E>,
+        reactor: reactor::Core,
+    }
+
+    impl<Req, Resp, E> fmt::Debug for Client<Req, Resp, E>
+        where Req: Serialize + 'static,
+              Resp: Deserialize + 'static,
+              E: Deserialize + 'static
+    {
+        fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+            write!(f, "Client {{ .. }}")
+        }
+    }
+
+    impl<Req, Resp, E> Client<Req, Resp, E>
+        where Req: Serialize + Sync + Send + 'static,
+              Resp: Deserialize + Sync + Send + 'static,
+              E: Deserialize + Sync + Send + 'static
+    {
+        /// Drives an RPC call for the given request.
+        pub fn call(&mut self, request: Req) -> Result<Resp, ::Error<E>> {
+            self.reactor.run(self.inner.call(request))
+        }
+    }
+
+    /// Extension methods for Clients.
+    pub trait ClientExt: Sized {
         /// Connects to a server located at the given address.
-        fn connect<A>(addr: A, options: Options) -> Result<Self, io::Error> where A: ToSocketAddrs;
+        fn connect<A>(addr: A, options: Options) -> io::Result<Self> where A: ToSocketAddrs;
+        /// Tries to clone the client.
+        fn try_clone(&self) -> io::Result<Self>;
```

Comment:
Done. Annnnnd it doesn't work, because the cloned client's internals still have a handle to the original reactor. So...it seems there's no way to clone a client onto a different reactor.